docs/proposals/algorithms/multi-edge-inference/proposal.md

```diff
@@ -0,0 +1,104 @@
+
+### Goals
+
+The goals of the multi-task joint inference for Ianvs are:
+
+- Adding an example named multi-task joint inference in Ianvs 
+  - choose tasks to realize joint inference according to various scenarios dynamically
+    - the characteristics of vehicle movement, the scenarios faced by autonomous vehicles are complicated
+  - fuse the output of multi tasks
+- Verifying the effectiveness of the joint inference in Object Detection
+
+## Proposal
+### Scope
+
+- Providing a new example for Ianvs
+
+## Design Details
+### Implementation
+
+Task, A triplet: Task Allocation +  Task samples + Task model 
+
+The architecture is shown in the below figure. The red box shows the work we are going to do. 
+
+<img src="Fig2_Architecture.png" width="600" />
+
+​																						Fig 2. Architecture
+
+
+- **Metaknowledge Initialization:** generates metaknowledge given the metadata base and training samples using meta learning
+
+
+- **Unseen Task Detection:** detects if the targeted task is covered in metaknowledge base
+- **Unseen Task Processing:**  updates the metaknowledge in the metaknowledge base 
+- **Task Ranking:**  ranks tasks according to the similarity
+
+  - During the process of inference, we will use a model which trained based metaknowledge to rank the tasks
+
+
+- **Task Allocation:** assigns tasks for the sample 
+
+  - After get the ranks of the tasks, we will first select some tasks to meet the time delay
+  - Then, according to the output of the model, wo use a threshold to determine the numbers of the tasks
+
+
+- **Joint Task Inference**:  aggregates the inference results of multiple allocated tasks
+
+
+  - In the Object Detection task, we first add all the bounding boxes together. The we use non maximum suppression(NMS) to get the final output. The figure 3 shows the process.
+
+    <img src="Fig3_Joint inference in Object Detection.png" width="600" />
+
+    ​                                                 Fig 3. Joint inference in Object Detection
+
+### Customize algorithm
+
+We want to realize it in Ianvs first. We need to submit the code in examples. The path of Ianvs of our example is as follows：
+
+``` 
+ianvs
+│  ...    
+└─examples
+│   └─bdd
+│       └─multi-task inference
+│           └─testalgorithms
+│           └─testenv
+│                 │testenv.yaml      #dataset、model_env、metrics
+│           │benchmarkingjob.yaml
+│  ... 
+```
+To add a new example in Ianvs, we need to prepare the test environments, provide the algorithms and give test reports or leaderboards. To prepare the test environments, we need to modify the testenv.yaml. The code of the algorithms is in  the folder: testalgorithms. The flow chart of the algorithms is in figure 4. 
+
+<img src="Fig4_algorithm in Ianvs.png" width="600" />
```

Comment:
Also, this flow chart can be intergated with ianvs modules of lifelong learning.